Android/Self_Assessment_Devarshi/Advanced/app/src/main/java/com/example/advanced/api/ApiClient.java

```diff
@@ -0,0 +1,80 @@
+package com.example.advanced.api;
+
+import com.example.advanced.InterfaceAdapter;
+import com.example.advanced.models.News;
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+
+import java.security.cert.CertificateException;
+
+import javax.net.ssl.HostnameVerifier;
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.SSLSession;
+import javax.net.ssl.SSLSocketFactory;
+import javax.net.ssl.TrustManager;
+import javax.net.ssl.X509TrustManager;
+
+import okhttp3.OkHttpClient;
+import retrofit2.Retrofit;
+import retrofit2.converter.gson.GsonConverterFactory;
+public class ApiClient {
+
+    public static final String BASE_URL = "https://newsapi.org/v2/";
+    public static Retrofit retrofit;
+
+    public static Retrofit getApiClient(){
+
+        if (retrofit == null){
+
+            retrofit = new Retrofit.Builder().baseUrl(BASE_URL)
+                    .client(getUnsafeOkHttpClient().build())
+                    .addConverterFactory(GsonConverterFactory.create())
+                    .build();
+
+        }
+
+        return retrofit;
+    }
+
+    public static OkHttpClient.Builder getUnsafeOkHttpClient(){
+        try {
+            // Create a trust manager that does not validate certificate chains
+            final TrustManager[] trustAllCerts = new TrustManager[]{
+                    new X509TrustManager() {
+                        @Override
+                        public void checkClientTrusted(java.security.cert.X509Certificate[] chain, String authType) throws CertificateException {
```

Comment:
The declared exception 'CertificateException' is never thrown, please resolve it.

Reply:
doubtful!
removed the thrown exception